src/backend/tests/projects.test.ts

```diff
@@ -925,6 +925,29 @@ describe('Projects', () => {
     });
   });
 
+  describe('Deleting unit', () => {
+    test('Delete unit does not work if user is not an admin or head', async () => {
+      await expect(ProjectsService.deleteUnit(theVisitor, 'NERSoftwareTools')).rejects.toThrow(
+        new AccessDeniedException('Only heads and above can delete a unit')
+      );
+    });
+
+    test('Deleting unit fails if unit does not exist', async () => {
+      vi.spyOn(prisma.unit, 'findUnique').mockResolvedValue(null);
+
+      await expect(ProjectsService.deleteUnit(batman, 'New Unit')).rejects.toThrow(
+        new NotFoundException('Unit', 'New Unit')
+      );
+    });
+
+    test('Delete Unit works', async () => {
+      vi.spyOn(prisma.unit, 'findUnique').mockResolvedValue(toolMaterial);
+      vi.spyOn(prisma.unit, 'delete');
+      const deletedUnit = await ProjectsService.deleteUnit(superman, 'NERSoftwareTools');
+      expect(prisma.unit.delete).toBeCalledTimes(1);
```

Comment:
to ensure that it is deleted, try to find the unit and ensure that it gives you a notfoundexception